hw/ip/lc_ctrl/doc/_index.md

```diff
@@ -5,14 +5,243 @@ title: "Life Cycle Controller Technical Specification"
+
+RAW_UNLOCK is not exposed in the open source design, rather it is something provisioned by the silicon creators prior to tapeout.
+It is the only token among those listed that is a global constant and stored in gates.
+
+All others CAN be device unique and are stored in OTP.
+
+### Token Usage Limits
+
+For conditional transitions, there is a limit to how many times they can be attempted.
+This is to prevent an attacker from brute-forcing any specific token, as this also helps to reduce the overall required token size.
+
+For OpenTitan, the total amount of state transitions and transition attempts is limited to 16.
+Once this number is reached, the life cycle controller rejects further attempts, effectively locking the device into its current state.
+
+The token counters are maintained in the OTP.
+To ensure the security of token limits cannot be bypassed, each request for a conditional transition **FIRST** increments the token count, and **THEN** checks for the validity of the token.
+
+### Token Hashing Mechanism
+
+All 128bit lock and unlock tokens are passed through a cryptographic one way function in hardware before the life cycle controller compares them to the provisioned values in OTP or to the netlist constant in case of RAW_UNLOCK.
+
+This mechanism is used to guard against reverse engineering and brute-forcing attempts.
+An attacker able to extract the hashed token values from the scrambled OTP partitions or from the netlist would first have to find a hash collision in order to perform a life cycle transition, since the values supplied to the life cycle controller must be valid hash pre-images.
+
+The token hashing mechanism leverages the PRESENT cipher primitive in order to construct a 128bit -> 128bit hashing function via a Merkle-Damgard construction, as described in the [OTP specification]({{< relref "hw/ip/otp_ctrl/doc/_index.md#unlock-token-hashing" >}}).
+
+### Post Transition Handling
+
+After a transition request, whether it was unconditional or conditional, the life cycle controller always disables all of its decoded outputs and puts the system in an inert state.
+The device is then expected to reboot before returning to a functional state.
+
+Note this happens for either successful or unsuccessful transitions.
+This general policy places a time-bound on how quickly life cycle states can change and also forces the device to behave more predictably.
+
+## Security Escalation
+
+The life cycle controller contains two escalation paths that are connected to [escalation severities 1 and 2 of the alert handler]({{< relref "hw/ip/alert_handler/doc/_index.md#escalation-signaling" >}}).
+
+The first escalation path is used to trigger escalation mechanisms such as secret wiping devices in OpenTitan.
+Upon assertion, the life cycle controller asserts the ESCALATE_EN life cycle signal that is distributed to all IPs in the design that implement an escalation action.
```

Comment:
that implement -> that implemented

Reply:
hm... I think present tense is fine here.
but let me reformulate this a bit since the sentence uses "that" twice right after another.